File: Backend/NoteBoard.Application/Models/Note/NoteViewModel.cs
using System;
using NoteBoard.Domain.Enumerators;

namespace NoteBoard.Application.Models.Note
{
    public sealed record NoteViewModel(int Id, string Title, string Content, bool IsCompleted, NoteColor Color, DateTimeOffset CreatedAt, int CreatedBy, DateTimeOffset? UpdatedAt);
}

File: Frontend/NoteBoardApp/src/app/pages/notes/models/note-view-model.ts
import { NoteColor } from './note-color';

export interface NoteViewModel {
  id: number;
  title: string;
  content: string;
  isCompleted: boolean;
  color: NoteColor;
  createdAt: string;
  createdBy: number;
  updatedAt: string | null;
}
